tests/bb_getopt_long.test/runit

```diff
@@ -0,0 +1,202 @@
+    test_and_compare noarg11-4 loosearg1 loosearg2 loosearg3 --noarg11 -p
+    test_and_compare noarg11-5 loosearg1 loosearg2 loosearg3 -p --noarg11
+    test_and_compare noarg11-6 loosearg1 loosearg2 -p loosearg3 --noarg11
+    test_and_compare noarg11-7 loosearg1 -p loosearg2 loosearg3 --noarg11
+    test_and_compare noarg11-8 -p loosearg1 loosearg2 loosearg3 --noarg11
+
+    # All args
+    test_and_compare various-1 loosearg1 loosearg2 --rarg0 arg0 --rarg1 arg1 --rarg2 arg2 --noarg3 --noarg4 --optarg6 --rarg7 arg7 --rarg8 arg8 --optarg9 --optarg10 --noarg11 --noarg12 -p
+    test_and_compare various-2 loosearg1 --rarg0=arg0 loosearg2 --rarg1 arg1 --rarg2=arg2 --noarg4 --optarg5 --optarg6 --rarg7 arg7 --rarg8 arg8 --optarg9 --optarg10 --noarg11 --noarg12 -p
+    test_and_compare various-3 loosearg1 --rarg0=arg0 --rarg1 arg1 loosearg2 --rarg2=arg2 --noarg3 --optarg5=arg5 --optarg6 --rarg7 arg7 --rarg8 arg8 --optarg9=arg9 --optarg10 --noarg12 -p
+    test_and_compare various-4 --rarg0=arg0 --rarg1 arg1 loosearg1 --rarg2=arg2 loosearg2 --noarg3 --optarg5=arg5 --optarg6 --rarg7 arg7 --rarg8 arg8 --optarg9=arg9 --optarg10 --noarg12 -p
+    test_and_compare various-5 --rarg0=arg0 --rarg1 arg1 loosearg1 loosearg2 --rarg2 arg2 --noarg3 --noarg4 --optarg5 --optarg6 --rarg7 arg7 --rarg8 arg8 --optarg9=arg9 --noarg12 -p
+    test_and_compare various-6 --rarg0=arg0 --rarg1 arg1 --rarg2 arg2 --noarg3 --noarg4 --optarg5 --optarg6 --rarg8=arg8 loosearg1 --optarg9=arg9 --optarg10 --noarg12 -p loosearg2 loosearg3
+    test_and_compare various-7 --rarg0=arg0 --rarg1 arg1 --rarg2 arg2 --noarg3 --noarg4 --optarg5 --optarg6 --rarg8=arg8 --optarg9=arg9 --optarg10 --noarg12 -p loosearg1 loosearg2 loosearg3
+    test_and_compare various-8 -p loosearg1 loosearg2 loosearg3 --rarg0=arg0 --rarg1 arg1 --rarg2 arg2 --noarg3 --noarg4 --optarg5 --optarg6 --rarg8=arg8 --optarg9=arg9 --optarg10
+    test_and_compare various-9 loosearg1 loosearg2 loosearg3 --rarg0 arg0 --rarg2=arg2 --noarg3 --noarg4 --optarg5 --optarg6 --rarg8=arg8 --optarg9=arg9 --optarg10 -p
+}
+
+function transform_args
+{
+    run_glgetopt transform-1 loosearg1 loosearg2 --rarg0 arg0 --rarg1 arg1 --rarg2 arg2 --noarg3 --noarg4 --optarg6 --rarg7 arg7 --rarg8 arg8 --optarg9 --optarg10 --noarg11 --noarg12 -p
+    run_bbgetopt transform-1 loosearg1 loosearg2 -rarg0 arg0 -rarg1 arg1 -rarg2 arg2 -noarg3 -noarg4 -optarg6 -rarg7 arg7 -rarg8 arg8 -optarg9 -optarg10 -noarg11 -noarg12 -p
+    diff transform-1.bb transform-1.gl
+    [[ $? != 0 ]] && error=1
+
+    run_glgetopt transform-2 --rarg0=arg0 loosearg1 --rarg1=arg1 --rarg2 arg2 --noarg3 loosearg2 --noarg4 --optarg6 --rarg7 arg7 --rarg8 arg8 --optarg9=arg9 --optarg10=arg10 -p
+    run_bbgetopt transform-2 -rarg0=arg0 loosearg1 -rarg1=arg1 --rarg2 arg2 --noarg3 loosearg2 --noarg4 --optarg6 --rarg7 arg7 --rarg8 arg8 -optarg9=arg9 -optarg10=arg10 -p
+    diff transform-2.bb transform-2.gl
+    [[ $? != 0 ]] && error=1
+
+
+}
+
```

Comment:
From Akshat: make sure that bb_getopt_long fails correctly - not the same exact string, but produces a failure on incorrect input